signer/signer.go

```diff
@@ -98,7 +102,9 @@ func (cfg *Configuration) GetPrivateKey() (crypto.PrivateKey, error) {
 }
 
 // ParsePrivateKey takes a PEM blocks are returns a crypto.PrivateKey
-func ParsePrivateKey(keyPEMBlock []byte) (crypto.PrivateKey, error) {
+// It tries to parse as many known key types as possible before failing and
+// returning all the errors it encountered.
```

Comment:
Any reason to do this instead of returning the first error?

Reply:
there's a lot of trial and error happening in that function. We start with the most common type and continue parsing until it succeeded or we give it. In that last case, I think returning all the errors is the best option to help debugging, but the formatting is also really terrible so I'm open to any better idea.

Reply:
Cool just wondering why.